guide.md

```diff
@@ -531,138 +533,87 @@ Which are all necessary changes. You may still want to try the GLFW based varian
-        if(key == GLFW_KEY_D)
-            keyState.d = false;        
-        if(key == GLFW_KEY_E)
-            keyState.e = false;        
-        if(key == GLFW_KEY_Q)
-            keyState.q = false;
-        if(key == GLFW_KEY_SPACE)
-            keyState.space = false;
-    }
+auto drag = ImGui::GetMouseDragDelta(1);
+```
+Note the 1 in the argument to get the drag delta for the _right_ mouse button. If the drag distance was zero we skip adding a force to an object, e.g., if this is done during a loop over all objects:
+```cpp
+if(ImGui::IsMouseReleased(ImGuiMouseButton_Right)){   
+    auto drag = ImGui::GetMouseDragDelta(1);
+    if(drag.x == 0 && drag.y == 0)
+        continue;
+    //...
 }
 ```
 
-Which is repetetive and error prone so try avoiding code like this if possible. We then modify our simulateStep function to no longer automatically rotate the cube and shoot a sphere when the spacebar is held down:
+To now calculate the mouse force we multiply the drag in the x-direction with the _right_ vector of the camera and the drag in y-direction with the _up_ vector of the camera. Due to conventions and preferences some users may prefer inverting certain components, e.g., should the mouse being moved left move the object to the left (this makes a lot of sense for dragging interactions) or should we use a _spring_ like interaction where we move the mouse to the right and on releasing the button the distance we moved influences the force applied to the object in the opposite direction (this makes a lot of sense for on release interactions). In a full game engine it makes sense to make this configurable for the user but here we invert the y-coordinate (due to screen coordinate systems) and keep the x as is:
 ```cpp
-// Scene1.cpp
-void Scene1::simulateStep(){
-    // pitch += pitch_increment;
-    // roll += roll_increment;
-    // yaw += yaw_increment;
-
-    glm::vec3 gravityAccel = glm::vec3(0, -9.81f, 0);
-
-    for (auto& particle : particles){
-        particle.position += 0.01f * particle.velocity;
-        particle.lifetime += 0.01f;
-        particle.velocity += gravityAccel * 0.01f;
-    }
-    
-    particles.erase(std::remove_if(particles.begin(), particles.end(), [](const Particle& particle){
-        return particle.lifetime > 1.f;
-    }), particles.end());
-
-    if(keyState.space)
-        launchSphere();
-    if(keyState.w)
-        pitch += pitch_increment;
-    if(keyState.s)
-        pitch -= pitch_increment;
-    if(keyState.a)
-        roll += roll_increment;
-    if(keyState.d)
-        roll -= roll_increment;
-    if(keyState.q)
-        yaw += yaw_increment;
-    if(keyState.e)
-        yaw -= yaw_increment;
-    lastLaunch++;
+if(ImGui::IsMouseReleased(ImGuiMouseButton_Right)){   
+    auto drag = ImGui::GetMouseDragDelta(1);
+    if(drag.x == 0 && drag.y == 0)
+        continue;
+    auto dx = drag.x * right;
+    auto dy = -drag.y * up;
+    object.force += dx + dy;

```

Comment:
1. I think it would be helpful if we added the hint that this section of code should be in the `Scene1::simulateStep()` method (similar to where we would have the button interaction as well)
2. I can imagine it would also be helpful if it showed how the force could act on the particles. Something simple would be
```cpp
if(ImGui::IsMouseReleased(ImGuiMouseButton_Right)){   
    auto drag = ImGui::GetMouseDragDelta(1);
    if(drag.x == 0 && drag.y == 0)
        continue;
    auto dx = drag.x * right;
    auto dy = -drag.y * up;
    auto force += dx + dy;
    for (auto& particle : particles){
        particle.position += force * 0.01f;
     }
}
```
This also shows that the force must be further discounted with the time step size.